{"ast":null,"code":"import axios from 'axios';\nimport { toast } from 'react-toastify';\naxios.interceptors.response.use(null, error => {\n  const expectedError = error.response && error.response.status >= 400 && error.response.status < 500;\n\n  if (!expectedError) {\n    console.log('Logging the error', error);\n    toast.error(\"An unexpected error occurred.\", {\n      position: \"top-center\",\n      autoClose: 5000,\n      hideProgressBar: false,\n      closeOnClick: false,\n      pauseOnHover: true,\n      draggable: true,\n      progress: undefined\n    });\n  }\n\n  return Promise.reject(error);\n});\nexport default {\n  get: axios.get,\n  post: axios.post,\n  delete: axios.delete,\n  put: axios.put\n};","map":{"version":3,"sources":["/home/ali/p/Programing-Training/React/http-app/src/services/httpService.js"],"names":["axios","toast","interceptors","response","use","error","expectedError","status","console","log","position","autoClose","hideProgressBar","closeOnClick","pauseOnHover","draggable","progress","undefined","Promise","reject","get","post","delete","put"],"mappings":"AAAA,OAAOA,KAAP,MAAkB,OAAlB;AACA,SAASC,KAAT,QAAsB,gBAAtB;AAEAD,KAAK,CAACE,YAAN,CAAmBC,QAAnB,CAA4BC,GAA5B,CAAgC,IAAhC,EAAsCC,KAAK,IAAI;AAE3C,QAAMC,aAAa,GAAGD,KAAK,CAACF,QAAN,IAAkBE,KAAK,CAACF,QAAN,CAAeI,MAAf,IAAyB,GAA3C,IAAkDF,KAAK,CAACF,QAAN,CAAeI,MAAf,GAAwB,GAAhG;;AAEA,MAAI,CAACD,aAAL,EAAoB;AAChBE,IAAAA,OAAO,CAACC,GAAR,CAAY,mBAAZ,EAAiCJ,KAAjC;AACAJ,IAAAA,KAAK,CAACI,KAAN,CAAY,+BAAZ,EAA6C;AAC3CK,MAAAA,QAAQ,EAAE,YADiC;AAE3CC,MAAAA,SAAS,EAAE,IAFgC;AAG3CC,MAAAA,eAAe,EAAE,KAH0B;AAI3CC,MAAAA,YAAY,EAAE,KAJ6B;AAK3CC,MAAAA,YAAY,EAAE,IAL6B;AAM3CC,MAAAA,SAAS,EAAE,IANgC;AAO3CC,MAAAA,QAAQ,EAAEC;AAPiC,KAA7C;AASH;;AAED,SAAOC,OAAO,CAACC,MAAR,CAAed,KAAf,CAAP;AACH,CAlBD;AAqBA,eAAe;AACXe,EAAAA,GAAG,EAAEpB,KAAK,CAACoB,GADA;AAEXC,EAAAA,IAAI,EAAErB,KAAK,CAACqB,IAFD;AAGXC,EAAAA,MAAM,EAAEtB,KAAK,CAACsB,MAHH;AAIXC,EAAAA,GAAG,EAAEvB,KAAK,CAACuB;AAJA,CAAf","sourcesContent":["import axios from 'axios';\nimport { toast } from 'react-toastify';\n\naxios.interceptors.response.use(null, error => {\n\n    const expectedError = error.response && error.response.status >= 400 && error.response.status < 500;\n\n    if (!expectedError) {\n        console.log('Logging the error', error);\n        toast.error(\"An unexpected error occurred.\", {\n          position: \"top-center\",\n          autoClose: 5000,\n          hideProgressBar: false,\n          closeOnClick: false,\n          pauseOnHover: true,\n          draggable: true,\n          progress: undefined,\n        });\n    }\n\n    return Promise.reject(error)\n});\n\n\nexport default {\n    get: axios.get,\n    post: axios.post,\n    delete: axios.delete,\n    put: axios.put\n}"]},"metadata":{},"sourceType":"module"}